cmd/goal/application.go

```diff
@@ -195,33 +192,25 @@ func init() {
 	methodAppCmd.Flags().MarkHidden("app-arg") // nolint:errcheck
 }
 
-type appCallBytes struct {
-	Encoding string `codec:"encoding"`
-	Value    string `codec:"value"`
-}
-
-func newAppCallBytes(arg string) appCallBytes {
-	parts := strings.SplitN(arg, ":", 2)
-	if len(parts) != 2 {
-		reportErrorf("all arguments and box names should be of the form 'encoding:value'")
-	}
-	return appCallBytes{
-		Encoding: parts[0],
-		Value:    parts[1],
+func newAppCallBytes(arg string) logic.AppCallBytes {
+	appBytes, err := logic.NewAppCallBytes(arg)
+	if err != nil {
+		reportErrorf(err.Error())
 	}
+	return logic.AppCallBytes(appBytes)
```

Comment:
@algochoi It appears the highlighted cast is unnecessary.  Can you double check me?

```suggestion
	return appBytes
```

Reply:
@algochoi Thanks for addressing in https://github.com/algorand/go-algorand/pull/4183/commits/9cf837c089e0f007a26fc29a9dbc2848d2796379 - resolving.